### PR TITLE
Re-add windows extension builds for R

### PR DESCRIPTION
**Pull request by hannes**

### CODE REVIEW

File: .github/workflows/R.yml [hunk 1]
```diff
@@ -0,0 +1,55 @@
...
+
+env:
+  GH_TOKEN: ${{ secrets.GH_TOKEN }}
+
+jobs:
+  rstats-windows-extensions:
+    # Builds extensions for windows_amd64_rtools
+    name: R Package Windows (Extensions)
+    if: startsWith(github.ref, 'refs/tags/v') || github.ref == 'refs/heads/main'
+    needs: rstats-linux
```
Comment (samansmink) [inline .github/workflows/R.yml:19]: The needs is incorrect

**Review by samansmink — COMMENTED** [review]
Found one minor issue

I've started a test run on this without the needs and the if https://github.com/samansmink/duckdb/actions/runs/6479465501/job/17593038237

I think maybe we wait for that one to finish? If that passes we can merge this?

File: .github/workflows/R.yml (hunk 1 above)
Comment (hannes) [reply .github/workflows/R.yml:19]: good catch

**Comment by hannes** [comment]
Well that did not work :/

**Comment by samansmink** [comment]
tried again with another fix, its the autoload tests that I added that break it

**Comment by hannes** [comment]
@Mytherin I think this can go in, ideally we have those binaries for 0.6.1 R Windows then

**Pull request merged**